habitat/tasks/rearrange/rearrange_sensors.py

```diff
@@ -86,6 +87,65 @@ def get_observation(self, *args, observations, episode, **kwargs):
         return batch_transform_point(pos, T_inv, np.float32).reshape(-1)
 
 
+class PositionGpsCompassSensor(Sensor):
+    def __init__(self, *args, sim, task, **kwargs):
+        self._task = task
+        self._sim = sim
+        super().__init__(*args, task=task, **kwargs)
+
+    def _get_sensor_type(self, *args, **kwargs):
+        return SensorTypes.TENSOR
+
+    def _get_observation_space(self, *args, config, **kwargs):
+        n_targets = self._task.get_n_targets()
+        return spaces.Box(
+            shape=(n_targets * 2,),
+            low=np.finfo(np.float32).min,
+            high=np.finfo(np.float32).max,
+            dtype=np.float32,
+        )
+
+    def _get_positions(self) -> np.ndarray:
+        pass
+
+    def get_observation(self, task, *args, **kwargs):
+        pos = self._get_positions()
+        robot_T = self._sim.robot.base_transformation
+
+        rel_pos = batch_transform_point(pos, robot_T.inverted(), np.float32)
+
+        polar_pos = []
```

Comment:
Nit: We know the dimensions of the polar_pos np array to begin with. Couldn't we just fill it instead of having a list to array conversion?